Library/Homebrew/api.rb

```diff
@@ -54,9 +54,11 @@ def self.fetch_json_api_file(endpoint, target: HOMEBREW_CACHE_API/endpoint,
         --speed-time #{ENV.fetch("HOMEBREW_CURL_SPEED_TIME")}
       ]
 
-      insecure_download = (ENV["HOMEBREW_SYSTEM_CA_CERTIFICATES_TOO_OLD"].present? ||
-                           ENV["HOMEBREW_FORCE_BREWED_CA_CERTIFICATES"].present?) &&
-                          !(HOMEBREW_PREFIX/"etc/ca-certificates/cert.pem").exist?
+      insecure_download = ((!DevelopmentTools.ca_file_handles_most_https_certificates? ||
+                            ENV["HOMEBREW_FORCE_BREWED_CA_CERTIFICATES"].present?) &&
+                           !(HOMEBREW_PREFIX/"etc/ca-certificates/cert.pem").exist?) ||
+                          (!DevelopmentTools.curl_handles_most_https_certificates? &&
+                           !HOMEBREW_BREWED_CURL_PATH.exist?)
```

Comment:
This logic is now very gnarly 😅. Would be good to split into variables or even a new function which can have early returns/`if` statements inside it.

Reply:
It should be more readable now; I hope it's close to what you were getting at.